genum/gen/generate.go

```diff
@@ -53,6 +55,7 @@ func (g *Generate) Parse() error {
 	pkgScope := pkg.Types.Scope()
 	g.Values = make([]Values, len(g.Types))
 	g.Traits = make([]TraitDescs, len(g.Types))
+	g.ParsableTraits = make([]TraitDescs, len(g.Types))
```

Comment:
Instead of creating a separate list of trait descriptions, I think it might streamline things a bit more to simply add a Parsable flag to the TraitDescs struct, and set it accordingly in the `extractTraitDescs` method. like:

```
		tDesc := TraitDesc{
			Name:    traitName,
			TypeRef: typeRef,
			Parsable: slices.Contains(g.ParsableByTraits, traitName),
			Traits: []TraitInstance{
				{
					OwningValue:  firstV,
					variableName: name,
					value:        v.Val().ExactString(),
				},
			},
		}
```

Then your loop logic in the template looks pretty much the same except for an extra `{{- if $.TraitDesc.Parsable }} code {{- end}}` as you iterate through the traits.

-- 

Alternatively, (or really, in addition to the first part, but to simplify the template code) since the whole "parsing" bit is getting quite complicated in the template, we could pre-calculate all the switch cases for parsing, so there's only one list to go though. Totally your call, though the first might end up being easier, the second might give you better duplicate detection for parsable values, and now that I wrote that, maybe it could make the switch statements look nicer too (i.e. you could combine multiple cases like `case "us-central1", "gcpuscentral1": return GCPUSCentral1`).